shm/Map.cpp

```diff
@@ -0,0 +1,51 @@
+
+#include <iostream>
+
+#include "Map.hpp"
+#include "ShmExceptions.hpp"
+
+Map::Map() : mapXsize(defaultXsize), mapYsize(defaultYsize)
+{
+    randomIslandGenerator(defaultIslandsAmount);
+}
+Map::Map(int sizeY, int sizeX, int islandsAmount) : mapXsize(sizeY), mapYsize(sizeX)
+{
+    if (islandsAmount <= (sizeY * sizeX))
```

Comment:
what if sizeY or sizeX is negative value?

Reply:
+1